packages/grid/src/vaadin-grid-data-provider-mixin.js

```diff
@@ -519,19 +534,58 @@ export const DataProviderMixin = (superClass) =>
       return result;
     }
 
-    scrollToIndex(index) {
-      super.scrollToIndex(index);
-      if (!isNaN(index) && (this._cache.isLoading() || !this.clientHeight)) {
-        this.__pendingScrollToIndex = index;
+    /**
+     * Scroll to a specific row index in the virtual list. Note that the row index is
+     * not always the same for any particular item. For example, sorting or filtering
+     * items can affect the row index related to an item.
+     *
+     * The `indexes` parameter can be either a single number or multiple numbers.
+     * The grid will first try to scroll to the item at the first index on the top level.
+     * In case the item at the first index is expanded, the grid will then try scroll to the
+     * item at the second index within the children of the expanded first item, and so on.
+     * Each given index points to a child of the item at the previous index.
+     *
+     * @param indexes {...string} the index of the item
```

Comment:
```suggestion
     * @param indexes {...number} the index of the item
```
Also the TS definition doc for this param has diverged a bit and could be aligned.